### PR TITLE
chore(node): update support to node v22

### PR DESCRIPTION
**Pull request by alicewriteswrongs**
Updating to support node v22

## What is the new behavior?

it's node v22! you can see some details about what's included in the release here: https://nodejs.org/en/blog/announcements/v22-release-announce

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Testing

If all of our automated tests are passing and you can build a sample project without issues then this should be good!

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -45,6 +45,7 @@
     },
     "pretty": true,
     "resolveJsonModule": true,
+    "skipLibCheck": true,
```
Comment (christian-bromann) [inline tsconfig.json:48]: Are there any TS issues introduced with this update?

Comment (alicewriteswrongs) [reply tsconfig.json:48]: yeah I saw an error where [path-scurry](https://www.npmjs.com/package/path-scurry) doesn't play nicely with the node typings:

```
node_modules/path-scurry/dist/commonjs/index.d.ts:117:31 - error TS2720: Class 'PathBase' incorrectly implements class 'Dirent'. Did you mean to extend 'Dirent' and inherit its members as a subclass?
  Property 'parentPath' is missing in type 'PathBase' but required in type 'Dirent'.

117 export declare abstract class PathBase implements Dirent {
                                  ~~~~~~~~

  node_modules/@types/node/fs.d.ts:247:9
    247         parentPath: string;
                ~~~~~~~~~~
    'parentPath' is declared here.


Found 1 error in node_modules/path-scurry/dist/commonjs/index.d.ts:117
```

Comment (alicewriteswrongs) [reply tsconfig.json:48]: I kind of like having `skipLibCheck` on because it does skip a bunch of type-checking and makes builds faster

Comment (alicewriteswrongs) [reply tsconfig.json:48]: just tested and I think it makes calling `tsc` around 3 seconds faster to have the option on

Comment (christian-bromann) [reply tsconfig.json:48]: Makes sense to me 👍

**Review by christian-bromann — APPROVED** [review]
👍

**Review by tanner-reits — APPROVED** [review]

**Pull request merged**